arrow-inject-compiler-plugin/src/main/kotlin/arrow/inject/compiler/plugin/fir/resolution/checkers/call/FirAbstractCallChecker.kt

```diff
@@ -117,7 +117,8 @@ internal interface FirAbstractCallChecker : FirAbstractProofComponent, FirResolu
           val originalTypeArgumentIndex =
             originalFunction?.typeParameters?.indexOfFirst {
               val originalTypeParameter = it.toConeType()
-              originalTypeParameter.render() == coneType.render()
+              // TODO: comparing `renderReadable` is not a good idea
```

Comment:
```suggestion
              // TODO: comparing `renderReadable` is not a good idea because it may not consider type constraints and other potential unique elements of a type not reflected in the rendered string.
```